OperatorAny - prevent multiple terminal events

davidmoten opened the pull request:
As per discussion in #4242, if an operator maps an `onNext` emission to an `onError` emission downstream then it needs be defensive about an `onCompleted` being sent from upstream even if upstream has been unsubscribed.

Includes a unit test that failed on the original code.

davidmoten committed:
OperatorAny - prevent multiple terminal events

akarnokd commented on src/main/java/rx/internal/operators/OperatorAny.java:
```diff
@@ -64,7 +64,10 @@ public void onNext(T t) {
+                if (!done) {
+                    done = true;
+                    child.onError(e);
+                }
```
Call `RxJavaHooks.onError` so no exception is lost.

davidmoten commented:
- Updated with `RxJavaHooks.onError` 
- checked `done` at start of `onNext` 
- added two unit tests for `(onNext, onNext)` and one for `(onNext, onError)` cases
